src/types.ts

```diff
@@ -4,19 +4,35 @@
  */
 
 /* eslint-disable @typescript-eslint/no-empty-interface */
-
-import type * as express from 'express';
 import type * as http from 'http';
 import type * as httpProxy from 'http-proxy';
 import type * as net from 'net';
 import type * as url from 'url';
 
-export interface Request extends express.Request {}
-export interface Response extends express.Response {}
+/**
+ * @warn Hide `req.url`, as it's unsafe for use due to express
+ * mutation.
+ *
+ * {@link https://github.com/expressjs/express/issues/4854}
+ *
+ * Use getUrl(req) to get a x-server safe URL.
+ */
+type ExpressCompatibleIncomingMessage = Omit<http.IncomingMessage, 'url'>;
```

Comment:
👀 . Here's a big change since our last review. Dropping `url` out of `IncomingMessage`, as it's unsafe to reference anywhere due to express support